### PR TITLE
Rename "quorum" term to "split brain protection"

### PR DESCRIPTION
**Pull request by petrpleshachkov**
Introduced split brain protection concept to replace quorum one to make
it more explicit and unambiguous.

EE part: https://github.com/hazelcast/hazelcast-enterprise/pull/3117
https://hazelcast.atlassian.net/wiki/spaces/PM/pages/346193957/Rename+Quorum+to+SplitBrain+Detection

### CODE REVIEW

**Comment by petrpleshachkov** [comment]
run-lab-run

**Comment by petrpleshachkov** [comment]
@blazember I've reviewed the comments and adjusted the code accordingly. Please let me know if you have more comments.

File: hazelcast/src/main/java/com/hazelcast/splitbrainprotection/SplitBrainProtectionOn.java [hunk 1]
```diff
@@ -14,25 +14,25 @@
...
 
-package com.hazelcast.quorum;
+package com.hazelcast.splitbrainprotection;
 
 /**
- * Represents a cluster quorum type
+ * Represents a cluster split brain protection type
  */
-public enum QuorumType {
+public enum SplitBrainProtectionOn {
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/splitbrainprotection/SplitBrainProtectionOn.java]: this may be called `SplitBrainProtectioType`.

Comment (petrpleshachkov) [reply hazelcast/src/main/java/com/hazelcast/splitbrainprotection/SplitBrainProtectionOn.java]: In the configuration we have renamed `type` -> `protectOn` and this is why I've decided to rename it here as well. O know type might look better here but being consistent with the configuration seems more important.

**Comment by petrpleshachkov** [comment]
Guys, thank you very much for the review. I am merging the PR.

**Pull request merged**